R/utils.R

```diff
@@ -129,31 +129,39 @@ render_dependencies <- function(dependencies, local = TRUE) {
     } else {
       "file"
     }
+    
+    # According to Dash convention, label react and react-dom as originating
+    # in dash_renderer package, even though all three are currently served
+    # up from the DashR package
+    if (dep$name %in% c("react", "react-dom")) {
+      dep$name <- "dash_renderer"
+    }
+
+    # The following lines inject _dash-component-suites into the src tags,
+    # as this is the current Dash convention. The dependency paths cannot
+    # be set solely at component library generation time, since hosted
+    # applications should have the app name injected as well.
+    #
+    # This is essentially analogous to this codeblock on the Python side:
+    # https://github.com/plotly/dash/blob/1249ffbd051bfb5fdbe439612cbec7fa8fff5ab5/dash/dash.py#L207
+    if (is.null(prefix)) {
+      dep[["script"]] <- paste0("/_dash-component-suites/",
+                                dep$name,
+                                "/",
+                                basename(dep[["script"]]))
+    } else {
+      dep[["script"]] <- paste0(sub("/", "", prefix), 
+                                "_dash-component-suites/", 
+                                dep$name,
+                                "/",
+                                basename(dep[["script"]]))
+    }
+    dep$src$file <- ""
     htmltools::renderDependencies(list(dep), src, encodeFunc = identity)
```

Comment:
it would be good to get the query string parameter in here `?v=<version-of-the-package>` so that caching works. It sounded like this was tough to do with `htmltools`, so I'd recommend just constructing these yourself with `sprintf` or w/e
It's a simple string, it's either e.g. 
```
<script type="text/javascript" src="/_dash-component-suites/dash_core_components/dash_core_components.min.js?v=0.43.1"></script>
```
for javascript files
or e.g. 
```
<link rel="stylesheet" href="/_dash-component-suites/dash_core_components/tabs.css?v=0.43.1">
```
for css files

Reply:
Definitely the way to go here, and would resolve some of the other issues you noted. Sorry for all the messy code, I'll fix this up and note the commit below.

Reply:
fixed in 2368c34deac2785e609563890d8862cd25041410

Reply:
I've opened a new issue to describe what's needed to fully :hocho: all references to `htmltools`: https://github.com/plotly/dashR/issues/54